File: public/docs/_examples/dependency-injection/dart/web/app.component.2.dart
// #docregion

// #docregion imports
import "package:angular2/core.dart";
import "car/car.component.dart";
import "heroes/heroes.component.1.dart";
import "package:angular2/core.dart";
import "app.config.dart";
import "logger.service.dart";

_provideAppConfigUseValueConfig() => provide("app.config", useValue: CONFIG);

// #enddocregion imports
@Component(
    selector: "my-app",
    template: '''
    <h1>{{title}}</h1>
    <my-car></my-car>
    <my-heroes></my-heroes>
  ''',
    directives: const [
      CarComponent,
      HeroesComponent
    ],
    providers: const [
      Logger,
      // #docregion provider-config
      _provideAppConfigUseValueConfig
    ])
class AppComponent {
  String title;
  // #docregion ctor
  AppComponent(@Inject("app.config") Config config) {
    title = config.title;
  }
}
// #enddocregion

File: public/docs/_examples/router-deprecated/ts/app/app.component.2.ts
/* Second Heroes version */
// #docplaster

// #docregion
import { Component }   from '@angular/core';
import { RouteConfig, ROUTER_DIRECTIVES } from '@angular/router-deprecated';

import { CrisisListComponent }   from './crisis-list.component';
// #enddocregion
/*
// Apparent Milestone 2 imports
// #docregion
// #docregion hero-import
import { HeroListComponent }     from './heroes/hero-list.component';
import { HeroDetailComponent }   from './heroes/hero-detail.component';
import { HeroService }           from './heroes/hero.service';
// #enddocregion hero-import
// #enddocregion
*/
// Actual Milestone 2 imports
import { HeroListComponent }     from './heroes/hero-list.component.1';
import { HeroDetailComponent }   from './heroes/hero-detail.component.1';
import { HeroService }           from './heroes/hero.service';
// #docregion

@Component({
  selector: 'my-app',
  template: `
    <h1>Component Router (Deprecated)</h1>
    <nav>
      <a [routerLink]="['CrisisCenter']">Crisis Center</a>
      <a [routerLink]="['Heroes']">Heroes</a>
    </nav>
    <router-outlet></router-outlet>
  `,
  providers:  [HeroService],
  directives: [ROUTER_DIRECTIVES]
})
// #enddocregion
/*
// #docregion route-config
@Component({ ... })
// #enddocregion route-config
*/
// #docregion
// #docregion route-config
@RouteConfig([
// #docregion route-defs
  {path: '/crisis-center', name: 'CrisisCenter', component: CrisisListComponent},
  {path: '/heroes',        name: 'Heroes',       component: HeroListComponent},
  // #docregion hero-detail-route
  {path: '/hero/:id',      name: 'HeroDetail',   component: HeroDetailComponent}
  // #enddocregion hero-detail-route
// #enddocregion route-defs
])
export class AppComponent { }
// #enddocregion route-config
// #enddocregion
